bin/utest/signal.c

```diff
@@ -295,6 +296,114 @@ TEST_ADD(signal_sigsuspend) {
   return 0;
 }
 
+/* ======= signal_sigtimedwait ======= */
+int sigtimedwait_child(void *arg) {
+  pid_t ppid = getppid();
+  syscall_ok(kill(ppid, SIGUSR1));
+  return 0;
+}
+
+TEST_ADD(signal_sigtimedwait) {
+  syscall_ok(signal(SIGCONT, sigcont_handler));
+  sigset_t set, current, waitset;
+  __sigemptyset(&set);
+  __sigaddset(&set, SIGUSR1);
+  __sigaddset(&set, SIGCONT);
+  syscall_ok(sigprocmask(SIG_SETMASK, &set, NULL));
+
+  utest_spawn(sigtimedwait_child, NULL);
+
+  siginfo_t info;
+  __sigemptyset(&waitset);
+  __sigaddset(&waitset, SIGUSR1);
+  assert(sigtimedwait(&waitset, &info, NULL) == SIGUSR1);
```

Comment:
NetBSD man:
> The assert() macro may be removed at compile time with the [cc(1)](https://man.netbsd.org/cc.1) option `-DNDEBUG.`

I think that you don't want to call anything inside assert.

Reply:
Thanks, didn't know that.

Reply:
@pj1031999 Notice, that in utests we use our own implementation of asserts that comes from `utest.h`.

Reply:
Right. There're still some confusing deficiencies of our userspace test framework. This is one of them. Personally I'd:
 * disallow using `assert` from by adding `-DNDEBUG` to `utest` Makefile,
 * rename our `assert` to `check`,
 * provide a set of functions with `x` prefix that guarantee to return with success as FreeBSD's [xmalloc](https://www.freebsd.org/cgi/man.cgi?query=xmalloc)